File: Programmers/[60057] 문자열 압축/woong-jae/60057.js
function solution(s) {
    let answer = s.length;
    
    const compress = (s, unit) => {
        const compressed = [];
        let lastChar = s.slice(0, unit), count = 1;
        
        for(let i = unit; i < s.length; i += unit) {
            const curChar = s.slice(i, i + unit);
            if(curChar === lastChar) {
                count++;
                continue;
            }
            if(count > 1) compressed.push(count);
            compressed.push(lastChar);
            lastChar = curChar;
            count = 1;
        }
        if(count > 1) compressed.push(count);
        compressed.push(lastChar);
        
        return compressed.join("");
    }
    
    for(let unit = 1; unit < s.length / 2 + 1; unit++) {
        const compressed = compress(s, unit);
        answer = Math.min(answer, compressed.length);
    }
    
    return answer;
}

File: Programmers/[60057] 문자열 압축/ONE/60057.java
class Solution {
    public int solution(String s) {
        int answer = Integer.MAX_VALUE;
        int len = s.length();

        for (int i = 1; i <= (len + 1) / 2; i++) {
            StringBuilder sb = new StringBuilder();
            for (int j = 0; j <= len - i; j += i) {
                String sub = s.substring(j, j + i);
                int count = 1;
                for (int k = j + i; k <= len - i; k += i) {
                    if (sub.equals(s.substring(k, k + i))) {
                        count++;
                        j = k;
                    }
                    else
                        break;
                }
                if(count == 1)
                    sb.append(sub);
                else
                    sb.append(count).append(sub);
            }
            sb.append(s.substring(len - (len % i)));
            answer = Math.min(answer, sb.length());
        }
        return answer;
    }
}